packages/server/lib/util/require_async.ts

```diff
@@ -0,0 +1,88 @@
+import _ from 'lodash'
+import * as path from 'path'
+import * as cp from 'child_process'
+import * as inspector from 'inspector'
+import * as util from '../plugins/util'
+import * as errors from '../errors'
+import Debug from 'debug'
+
+const debug = Debug('cypress:server:require_async')
+
+let requireProcess: cp.ChildProcess | null
+
+interface RequireAsyncOptions{
+  projectRoot: string
+  loadErrorCode: string
+  /**
+   * members of the object returned that are functions and will need to be wrapped
+   */
+  functionNames: string[]
+}
+
+interface ChildOptions{
+  stdio: 'inherit'
+  execArgv?: string[]
+}
+
+const killChildProcess = () => {
+  requireProcess && requireProcess.kill()
+  requireProcess = null
+}
+
+export default async function requireAsync (filePath: string, options: RequireAsyncOptions): Promise<any> {
```

Comment:
Any specific reason to use a default export? I think we generally should use named exports where possible, seems to be the pattern with all the new TS code.

Reply:
It was done this way in #17000 is the only reason.
I will "fix" it right away.

Reply:
Done in 13b603d